src/classes/RD_RecurringDonations_TEST.cls

```diff
@@ -501,19 +829,57 @@ public class RD_RecurringDonations_TEST {
         r1.npe03__Open_Ended_Status__c = system.label.npe03.RecurringDonationOpenStatus;
         r1.npe03__Next_Payment_Date__c = system.today().toStartOfMonth();
         insert r1;
-        
-        list<Opportunity> originalOpps = new list<Opportunity>([select id, Name,amount,accountid,CloseDate from Opportunity where npe03__Recurring_Donation__c = :r1.id order by CloseDate]);          
-        
-        integer cOpp =  originalOpps.size();
-        system.assert(cOpp == 52 || cOpp == 53);        
-        system.assertEquals(100, originalOpps[0].Amount);
+
+        List<Opportunity> originalOpps = [SELECT Id
+                                        , Name
+                                        , Amount
+                                        , AccountId
+                                        , CloseDate
+                                        , StageName
+                                        FROM Opportunity
+                                        WHERE npe03__Recurring_Donation__c = :r1.id];
+
+        Integer cOpp =  originalOpps.size();
+        //depending on the day of week of the start date, there could be 52 or 53 in a given calendar year
+        System.assert(cOpp == 52 || cOpp == 53);
+        System.assertEquals(100, originalOpps[0].Amount);
         Test.startTest();
         r1.npe03__Installment_Period__c = system.label.npe03.RecurringDonationInstallmentPeriodMonthly;
         update r1;
         Test.stopTest();
-        
-        system.assertEquals(12, [select count() from Opportunity where npe03__Recurring_Donation__c = :r1.id]);        
+
+        List<Error__c> errors = [SELECT Id, Full_Message__c FROM Error__c];
+        System.assertEquals(errors.size(),0,errors);
+
+        List<Opportunity> postUpdateOpps = [SELECT Id
+                                            , Name
+                                            , Amount
+                                            , AccountId
+                                            , CloseDate
+                                            , StageName
+                                            FROM Opportunity
+                                            WHERE npe03__Recurring_Donation__c = :r1.id];
+
+        System.assertEquals(postUpdateOpps.size(), originalOpps.size(), 'No opportunities should have been deleted.');
+
+        for(Integer i; i<originalOpps.size(); i++) {
+            System.assertEquals(originalOpps[i].Id, postUpdateOpps[i].Id, 'The original opportunity records should persist.');
+        }
+
+        // these two integers, 12 and 13, are likely fragile
+        // TODO: refactor with hard-coded start date so it's stable
```

Comment:
What does the TODO refer to here since the code above does use a fixed date for `Date_Established__c`?

Reply:
Thinking through this more, it is stable. It does use a dynamic npe03__Next_Payment_Date__c, but months are stable, whereas weeks are not, hence the || in line 844. Removed the comments.